contracts/encoders/evm-encoder/src/contract.rs

```diff
@@ -0,0 +1,67 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+
+use cosmwasm_std::{
+    to_json_binary, Binary, Deps, DepsMut, Empty, Env, MessageInfo, Response, StdResult,
+};
+use cw2::set_contract_version;
+use valence_encoder_utils::msg::{ProcessorMessageToEncode, QueryMsg};
+
+use crate::{evict_msgs, insert_msgs, pause, resume, send_msgs, EVMLibrary};
+
+// version info for migration info
+const CONTRACT_NAME: &str = env!("CARGO_PKG_NAME");
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    _msg: Empty,
+) -> StdResult<Response> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    Ok(Response::new().add_attribute("method", "instantiate"))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(_deps: DepsMut, _env: Env, _info: MessageInfo, _msg: Empty) -> StdResult<Response> {
+    unimplemented!("This contract does not handle any execute messages, only queries")
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn query(_deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
+    match msg {
+        QueryMsg::IsValidLibrary { library } => to_json_binary(&is_valid_library(library)),
+        QueryMsg::Encode { message } => encode(message),
+    }
+}
+
+fn is_valid_library(library: String) -> bool {
```

Comment:
This will be used when the program owner creates an authorization with an EVM subroutine. Basically he will say what encoder/library we are targeting and I will verify that it's valid. To avoid people creating subroutines that will fail.

It's not a sanity check but something I know I'll eventually need. Therefore there is a query that uses this.

Reply:
ah I see, cool